spring-integration-core/src/main/java/org/springframework/integration/store/MessageGroupStore.java

```diff
@@ -140,14 +129,31 @@
 	MessageGroupMetadata getGroupMetadata(Object groupId);
 
 	/**
-	 * Return the one {@link org.springframework.messaging.Message} from {@link org.springframework.integration.store.MessageGroup}.
+	 * Return the one {@link Message} from {@link MessageGroup}.
 	 * @param groupId The group identifier.
-	 * @return the {@link org.springframework.messaging.Message}.
+	 * @return the {@link Message}.
 	 * @since 4.0
 	 */
 	Message<?> getOneMessageFromGroup(Object groupId);
 
 	/**
+	 * Store messages with an association to a group id.
+	 * This can be used to group messages together.
+	 * @param groupId The group id to store messages under.
+	 * @param messages The messages to add.
+	 * @since 4.3
+	 */
+	void addMessagesToGroup(Object groupId, Message<?>... messages);
```

Comment:
That's awry here because we can't have an overloaded method with different return type like an existing already. See `BasicMessageGroupStore`:
```
MessageGroup addMessageToGroup(Object groupId, Message<?> message);
```
I did like this to align with the:
```
void removeMessagesFromGroup(Object key, Collection<Message<?>> messages);
```

But I'm open for any other idea.

I'm really not going to deprecate `addMessageToGroup()`, BTW.